api/v1/mongodbcommunity_types.go

```diff
@@ -364,6 +358,42 @@ type MongoDBCommunity struct {
 	Status MongoDBCommunityStatus `json:"status,omitempty"`
 }
 
+// GetScramOptions returns a set of Options that are used to configure scram
+// authentication.
+func (m MongoDBCommunity) GetScramOptions() scram.Options {
+	return scram.Options{
+		AuthoritativeSet:   true,
+		KeyFile:            scram.AutomationAgentKeyFilePathInContainer,
+		AutoAuthMechanisms: []string{scram.Sha256},
+		AgentName:          scram.AgentName,
+		AutoAuthMechanism:  scram.Sha256,
+	}
+}
+
+// GetScramUsers converts all of the users from the spec into users
+// that can be used to configure scram authentication.
+func (m MongoDBCommunity) GetScramUsers() []scram.User {
+	var users []scram.User
```

Comment:
[nit] can we pre-allocate this and avoid `append`?

Reply:
(the same for roles below)

Reply:
good idea, will update!